packages/lib/models/Setting.ts

```diff
@@ -201,6 +201,22 @@ const defaultMigrations: DefaultMigration[] = [
 	},
 ];
 
+// "UserSettingMigration" are used to migrate existing user setting to a new setting. With a way
+// to transform existing value of the old setting to value and type of the new setting.
+interface UserSettingMigration {
+	oldName: string;
+	newName: string;
+	transformValue: Function;
+}
+
+const userSettingMigration: UserSettingMigration[] = [
+	{
+		oldName: 'spellChecker.language',
+		newName: 'spellChecker.languages',
+		transformValue: (value: string) => { return [value]; },
+	},
+];
```

Comment:
Very good - and we'll also need tests for this. See in Setting.test.ts the `applyDefaultMigrations()` tests.

Reply:
Added tests.